### PR TITLE
Fix trace tests

### PR DESCRIPTION
**Pull request by Julio-Guerra**
- APPSEC_KEEP didn't get approved so MANUAL_KEEP should be validated instead
- Check for every appsec span tags in case of an event, and not only the event-specific tags.
- Rework the test file.
- Add the missing user agent span tag in Go.

### CODE REVIEW

File: tests/appsec/test_traces.py [hunk 1]
```diff
@@ -15,9 +15,14 @@
...
 class Test_AppSecEventSpanTags(BaseTestCase):
     def test_appsec_event_span_tags(self):
-        """Spans with AppSec events should have the general AppSec span tags, along with the appsec.event and _sampling_priority_v1 tags"""
+        """
+        Spans with AppSec events should have the general AppSec span tags, along with the appsec.event and
+        _sampling_priority_v1 tags
+        """
 
         @missing_feature(library="ruby", reason="can't report user agent with dd-trace-rb")
+        @missing_feature(library="dotnet",
```
Comment (cbeauchesne) [inline tests/appsec/test_traces.py:24]: This decorator should be on the test function (`test_appsec_event_span_tags`), not here

Comment (Julio-Guerra) [reply tests/appsec/test_traces.py]: Oops, my bad, now fixed.

File: tests/appsec/test_traces.py [hunk 2]
```diff
@@ -33,42 +39,48 @@ def validate_appsec_span(span):
...
                 raise Exception("Metric _sampling_priority_v1 should be set on traces that are manually kept")
 
-            if span["metrics"]["_sampling_priority_v1"] != APPSEC_KEEP:
-                raise Exception(f"Trace id {span['trace_id']} , sampling priority should be {APPSEC_KEEP}")
+            MANUAL_KEEP = 2
+            if span["metrics"]["_sampling_priority_v1"] != MANUAL_KEEP:
+                raise Exception(f"Trace id {span['trace_id']} , sampling priority should be {MANUAL_KEEP}")
 
-            return True
+            return validate_appsec_span_tags(span)
```
Comment (cbeauchesne) [inline tests/appsec/test_traces.py:46]: Useles no? `test_custom_span_tags` test it for all span.

File: tests/appsec/test_traces.py [hunk 3]
```diff
@@ -33,42 +39,48 @@ def validate_appsec_span(span):
...
+
+        interfaces.library.add_span_validation(validator=validate_custom_span_tags)
+
+
+RUNTIME_FAMILY = ["nodejs", "ruby", "jvm", "dotnet", "go", "php", "python"]
+
 
-            if "_dd.appsec.enabled" not in span["metrics"]:
-                raise Exception("Can't find _dd.appsec.enabled in span's metrics")
+def validate_appsec_span_tags(span):
```
Comment (cbeauchesne) [inline tests/appsec/test_traces.py:68]: If my remark above is right, you can move this test logic inside `validate_custom_span_tags`

File: tests/appsec/test_traces.py (hunk 2 above)
Comment (Julio-Guerra) [reply tests/appsec/test_traces.py:46]: Those span tags are not checked otherwise. I would check them when an attack happens too.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**